PYME/LMVis/Extras/extra_layers.py

```diff
@@ -174,6 +174,20 @@ def save_surface(visFr):
         else:
             raise ValueError('Invalid file extension .' + str(ext))
     
+def distance_to_surface(visFr):
+    from PYME.recipes.surface_fitting import DistanceToSurface
+
+    dist_name = visFr.pipeline.new_ds_name('distance')
+    _, surf_count = visFr.pipeline.new_ds_name('surf', return_count=True)
```

Comment:
This is super gross! It will generate a new unused surface name each time it is called

Reply:
Would be much less gross to look at all the surfaces which are already in the pipeline and parse their names.

Reply:
Yeah, I'll make a new function in `pipeline` to do this. This isn't the first time I've run into this.

Reply:
Wait, will this generate a new unused surface name each time it's called? If I never create the surface using the generated name, I don't think it will.

Reply:
> Would be much less gross to look at all the surfaces which are already in the pipeline and parse their names.

I think this is what `new_ds_name` does. It just also returns the next name. However, if you don't use it, it will just return the same name/count the next time.

Reply:
Or are you proposing that for each surface in existence I calculate the distance to that surface?

Reply:
You're right - it does look to see if a surface exists before incrementing the count - I somehow thought the count was global and incremented on each call.

The best solution would probably be to do something like: 
```python
surface_names = [k for k, v in pipeline.dataSources.items() if isinstance(v, TriangleMesh)]
surf_name = wx.SingleChoiceDialog(surface_names)
```
potentially short-cutting if there is only one surface.

Relying on surface names following a `surfxx` pattern might be optimisitic.

Reply:
This is just an initial guess for a dataSource name, and not a bad one given how a user is expected to generate a surface (the surface name will follow the `surfxxx` pattern). The user selects the data source in `dts.configure_traits(kind='modal')` on line 187. I'm not sure creating a `wx.SingleChoiceDialog` just before the traits configuration makes sense. Is there a way to limit the list of choices for `input_surface` in `configure_traits` to instances of `TriangleMesh`?

Reply:
Restricting the choices might not be trivial, but is there any actual need to call configure_traits - the `DistanceToSurface` module doesn't have any parameters other than input and output names - we could use the current selected data source as the input points, prompt for the mesh, and generate a name for the output using `new_ds_name()`  - I suspect this would result in a better / easier UI overall.

Reply:
OK, I'm cool with that solution. Will push a commit changing to this shortly.